fix(accounting): fix incorrect output in Budget Variance Report

kevingdc opened the pull request:
This pull request fixes a bug in **Budget Variance Report** where actual expense amounts are added across different fiscal years. This was fixed by
updating the function and queries for computing the actual expense
amounts.

Expense entry:
<img width="1216" alt="journal_entry" src="https://user-images.githubusercontent.com/16395892/81286496-bdf89a80-9093-11ea-81d5-d1ba246b262c.png">

Before:
<img width="1188" alt="incorrect" src="https://user-images.githubusercontent.com/16395892/81286516-c2bd4e80-9093-11ea-9f2b-cbcc4d0edef9.png">

After:
<img width="1200" alt="correct" src="https://user-images.githubusercontent.com/16395892/81286531-c650d580-9093-11ea-939d-ef68601cd841.png">

kevin98gray committed:
style: Improve formatting

This commit improves indentations and makes sql queries more readable.

kevin98gray committed:
fix: Fix Budget Variance Report

This commit fixes a bug in Budget Variance Report where it combines the
actual expense amounts across different fiscal years. This was fixed by
updating the function and queries for computing the actual expense
amounts.

deepeshgarg007 commented on erpnext/accounts/report/budget_variance_report/budget_variance_report.py:
```diff
@@ -2,187 +2,419 @@
+                    tav_dict = cam_map[ccd.name][account][year][month]
 
-			for ad in actual_details.get(ccd.account, []):
-				if ad.month_name == month:
```
@kevingdc No need to make all these changes, a small change here would have fixed the issue. No need make extra queries to get budget_amount or monthly_distribution amount seperately.

`if ad.month == month and ad.fiscal_year == ccd.fiscal_year` this condition would have been enough. Also there are lot of inconsistencies in indentation spacing, somewhere spaces and somewhere tabs, we usually prefer tabs and that is what we have used in the entire project

kevingdc replied:
Hi @deepeshgarg007,

Thank you for the response. I appreciate the feedback, and you're right, adding the the condition `if ad.month == month and ad.fiscal_year == ccd.fiscal_year` would have fixed it. Although the reason why I added those functions was so I could also show the actual expense even though it doesn't have a budget allocated, an example is shown below:

<img width="1179" alt="Screen Shot 2020-05-08 at 4 55 26 PM" src="https://user-images.githubusercontent.com/16395892/81390049-674f9700-914d-11ea-87ef-7fb4d8a36c51.png">

The old code, even though it would get fixed by adding the condition you mentioned wouldn't have shown it. The output would have been like this:

<img width="1177" alt="Screen Shot 2020-05-08 at 5 00 58 PM" src="https://user-images.githubusercontent.com/16395892/81390111-80f0de80-914d-11ea-99d7-8774152ef2c4.png">


I thought it would be more helpful to the users to see their actual expenses even though there wasn't a budget allocated for it. But if you feel that this is unnecessary, please tell me and I'll update my code to reflect your request. 

Thank you!

deepeshgarg007 replied:
Budget Variance Report is to check variation of Actual Expense from Budget. This doesn't seems the right place to show expense if budget doesn't exits, also it adds a lot of unnecessary query to get this done and fix the spaces for indentation as welll

kevingdc replied:
@deepeshgarg007 got it. Updated the code to reflect this.

kevin98gray committed:
fix: Simplify get_dimension_account_month_map

This commit updates get_dimension_account_month_map to no longer show
the actual expense when there is no budget. This also removes the other
functions and queries related to it. Spaces are also converted to tabs.

deepeshgarg007 committed:
fix: Formatting fixes
